### PR TITLE
Display Version Year on Levels Page

### PR DESCRIPTION
**Pull request by epeach**
Adds "Version: <unit year>" on levels in the small footer. Screenshots below - overlapping is due to how I compressed the page to be able to show both the url and the footer and is not an artifact of this change.

Screenshot from 2022-10-20 14-39-09

Express Course. Highlights that the text wraps when compressed.
Screenshot from 2022-10-20 14-39-53

Screenshot from 2022-10-20 14-38-42

### CODE REVIEW

File: dashboard/app/views/layouts/_small_footer.html.haml [hunk 1]
```diff
@@ -12,7 +12,8 @@
     # false because we're not rendering into a phone wireframe here
     phoneFooter: false,
     fullWidth: full_width,
-    channel: channel
+    channel: channel,
+    scriptYear: view_options[:script_year]
```
Comment (hannahbergam) [inline dashboard/app/views/layouts/_small_footer.html.haml]: Yay this is very clean!! One thing I have no answers about- how much of the things in our codebase labeled 'script' should be converted to 'unit'?

**Review by hannahbergam — APPROVED** [review]
Is this something we could add a tiny test for? Or will it be more about approving a new eyes baseline?

File: dashboard/app/helpers/levels_helper.rb [hunk 2]
```diff
@@ -260,7 +260,7 @@ def app_options
     end
 
     if @script
-      view_options(script_name: @script.name)
+      view_options(script_name: @script.name, script_year: @script.get_course_version.key)
```
Comment (bethanyaconnor) [inline dashboard/app/helpers/levels_helper.rb]: Some of our course versions have a key "unversioned" which means that it was a one off unit (for example, for hour of code). It looks a little odd in the footer so maybe it's worth not displaying it if the key is "unversioned"? Up to you if you want this logic to be on the client or backend.

From http://localhost-studio.code.org:3000/s/dance-2019/lessons/1/levels/8:
Screen Shot 2022-10-21 at 12 44 08 PM

**Review by bethanyaconnor — COMMENTED** [review]
Love this approach! Just a comment about one edge case :)

File: dashboard/app/views/layouts/_small_footer.html.haml (hunk 1 above)
Comment (epeach) [reply dashboard/app/views/layouts/_small_footer.html.haml]: Oof, that's a great question. I'm going to ask in the team slack channel to see what the consensus is.

File: dashboard/app/helpers/levels_helper.rb (hunk 2 above)
Comment (epeach) [reply dashboard/app/helpers/levels_helper.rb]: Thanks for pointing out the edge case! Are there others we might run in to? I'm trying to decide if we would benefit from explicitly looking for 'unversioned' and not displaying in that case, or if we should disallow any string that isn't comprised of numbers (therefore only allowing 'years').

Comment (epeach) [reply dashboard/app/helpers/levels_helper.rb]: Updated to filter out all keys that are not entirely numbers. Hopefully leaving only years!
Screenshot from 2022-10-21 11-39-22

Comment (hannahbergam) [reply dashboard/app/helpers/levels_helper.rb]: I love this approach! Seems more future-proof:)

**Review by bethanyaconnor — APPROVED** [review]
Curious what you think about Hannah's question around testing. Part of me thinks that this will be on eyes diffs anyway so they might be sufficient testing?

**Comment by epeach** [comment]
@bethanyaconnor - I spent some time working on adding testing in the dashboard test for levels_helpers, but got stuck trying to figure out what I would be actually testing in there. The major outcome I see if that the version year either appears or doesn't appear, which will be caught in eyes from the logic in jsx. If we think a test here would give us confidence, I'd love to pair a little bit to get clarity on the expected coverage on that test(s)!

**Comment by bethanyaconnor** [comment]
> The major outcome I see if that the version year either appears or doesn't appear, which will be caught in eyes from the logic in jsx. 

That was my thought too -- that's enough confidence for me!

**Pull request merged**